src/test/mina_automation/executor.ml

```diff
@@ -11,83 +11,113 @@ module DockerContext = struct
     { image : string; workdir : string; volume : string; network : string }
 end
 
-type context =
-  | Dune (* application ran from dune exec command *)
-  | Local (* application ran from _build/default folder*)
-  | Debian (* application installed from mina debian package *)
-  | Docker of DockerContext.t
-  | AutoDetect
 (* application ran inside docker container *)
 
-module Executor = struct
-  type t = { official_name : string; dune_name : string; context : context }
+module type AppPaths = sig
+  val dune_name : string
 
-  let of_context ~context ~dune_name ~official_name =
-    { context; dune_name; official_name }
-
-  let run_from_debian t ~(args : string list) ?env () =
-    Util.run_cmd_exn ?env "." t.official_name args
+  val official_name : string
+end
 
-  let run_from_dune t ~(args : string list) ?env () =
-    Util.run_cmd_exn ?env "." "dune" ([ "exec"; t.dune_name; "--" ] @ args)
+let logger = Logger.create ()
 
-  let run_from_local t ~(args : string list) ?env () =
-    Util.run_cmd_exn ?env "."
-      (Printf.sprintf "_build/default/%s" t.dune_name)
-      args
+module Make_PathFinder (P : AppPaths) = struct
```

Comment:
At a first glance, this doesn't have to be a functor, just a type and some associated functions should be fine. But I assume there's some later modification making this necessary.

Reply:
Let me check on this

Reply:
Yes, I agree the original implementation is nicer. We could revert to that version, just I still think we should avoid this field that cause a functor to boom on construction when possible. 

```ocaml
  let standalone_path_exn =
    Deferred.map standalone_path ~f:(fun opt ->
        Option.value_exn opt
          ~message:
            "Could not find standalone path. App is not executable outside the \
             dune" )
```